ui/src/app/components/results/test-case-result-details.component.ts

```diff
@@ -373,4 +382,40 @@ export class TestCaseResultDetailsComponent extends BaseComponent implements OnI
     this.hasSteps = hasSteps;
     console.log(hasSteps);
   }
+
+  openDryRun() {
+    this.matModal.open(DryRunFormComponent, {
+      height: "100vh",
+      width: '60%',
+      position: {top: '0px', right: '0px'},
+      panelClass: ['mat-dialog', 'rds-none'],
+      data: {
+        testCaseId: this.testCase.id
+      },
+    })
+  }
+
+  public fetchVersion(workspaceVersionId:number){
+    this.chromeRecorderService.isChromeBrowser();
+    this.chromeRecorderService.pingRecorder();
+    this.workspaceVersionService.show(workspaceVersionId).subscribe((res)=>{
+      this.version = res;
+      if (this.version.workspace.isWebMobile) {
+        this.chromeRecorderService.isChromeBrowser();
+        this.chromeRecorderService.pingRecorder();
+        setTimeout(() => {
```

Comment:
why do we need setTimeout here?

Reply:
for the first time it is taking some time to ensure that recorder is installed for that we are using time out

Reply:
Ideally, when there is a ping happened, there should be an expectation of ack from the peer. It should be two way communication.

Reply:
Got it